### PR TITLE
feat: ✨ add sd-table and sd-table-cell

### PR DESCRIPTION
**Pull request by MarcMatthiae**
## Description:
Add css components sd-table and sd-table-cell.

## Definition of Reviewable:
- [x] Documentation is created/updated
- [x] Migration Guide is created/updated
- [x] Stories (features, a11y) are created/updated
- [x] relevant tickets are linked
- [x] PR is assigned to project board

### CODE REVIEW

File: packages/components/src/styles/table-cell/table-cell.stories.ts [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+export const Default = {
+  render: (args: any) => {
+    return generateTemplate({
+      options: { templateContent: '<td class="%CLASSES%">%SLOT%</td>' },
+      args
+    });
+  }
+};
+
+// TODO: Add sample "Simple Table, First Column Fixed" and "Multi Select Table" as soon as we have sd-scrollable.
```
Comment (karlbaumhauer) [inline packages/components/src/styles/table-cell/table-cell.stories.ts:38]: pls remove the ToDo and open up a ticket for this. ToDos in the code usually get lost... ;-)

**Review by karlbaumhauer — CHANGES_REQUESTED** [review]

**Comment by mariohamann** [comment]
CleanShot 2023-09-13 at 12 40 44@2x

Please make the background colors an enum, calling it `sd-table-cell--bg-...` to make clear, that this is only for the background.

See e. g. 
CleanShot 2023-09-13 at 12 42 34@2x

File: packages/components/src/styles/table-cell/table-cell.stories.ts [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+          margin: 0;
+          padding: 16px;
+          align-items: center;
+          justify-content: flex-start;
+          gap: 16px;
+          text-align: left;
+        }
+
+        .sd-table-cell.sortable button:hover {
+          background-color: white;
```
Comment (mariohamann) [inline packages/components/src/styles/table-cell/table-cell.stories.ts:146]: It would be helpful to make this visible in the story's code.

**Comment by MarcMatthiae** [comment]
> CleanShot 2023-09-13 at 12 40 44@2x
> Please make the background colors an enum, calling it `sd-table-cell--bg-...` to make clear, that this is only for the background.
> 
> See e. g. CleanShot 2023-09-13 at 12 42 34@2x

I understood, that we provide this as part of the table-cell component and we also wrote that in the story: 
image

I don´t know if it is necessary, but at least we refined it like that.

**Comment by mariohamann** [comment]
> > CleanShot 2023-09-13 at 12 40 44@2x
> > Please make the background colors an enum, calling it `sd-table-cell--bg-...` to make clear, that this is only for the background.
> > See e. g. CleanShot 2023-09-13 at 12 42 34@2x
> 
> I understood, that we provide this as part of the table-cell component and we also wrote that in the story: image
> 
> I don´t know if it is necessary, but at least we refined it like that.

Yeah, but we should document it properly. Just provide it as an own style. :)

File: packages/components/src/styles/table-cell/table-cell.css [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  }
+
+  &--neutral-100 {
+    @apply bg-neutral-100;
+  }
+}
+
+.sd-table-cell * {
+  @apply align-middle;
+}
```
[the remark this answers is not in the document]
Comment (mariohamann) [reply packages/components/src/styles/table-cell/table-cell.css:37]: But maybe it should. :)

[the remark this answers is not in the document]
Comment (MarcMatthiae) [reply packages/components/src/styles/table-cell/table-cell.css:37]: Actually in my samples this did not make a difference, so I can just remove it.

File: packages/components/src/styles/table-cell/table-cell.css [hunk 4]
```diff
@@ -0,0 +1,37 @@
+/* 
+* Remove inherited styles from table before applying our styles.
+*/
+.sd-table {
+  all: unset;
+  display: table;
```
Comment (yoezlem) [inline packages/components/src/styles/table-cell/table-cell.css:6]: Why did you not add the tailwind classes?

@apply table border-collapse border-spacing-0



<br class="Apple-interchange-newline">

Comment (MarcMatthiae) [reply packages/components/src/styles/table-cell/table-cell.css:6]: Good question...Will change this.

File: packages/components/src/styles/table-cell/table-cell.stories.ts (hunk 2 above)
Comment (yoezlem) [reply packages/components/src/styles/table-cell/table-cell.stories.ts:146]: Maybe we can also replace the CSS part with our tokens...?

Comment (MarcMatthiae) [reply packages/components/src/styles/table-cell/table-cell.stories.ts:146]: I wanted to do that, but I could not find out how to use our tokens in storybook. Do you know how to do that?

File: packages/components/src/styles/table-cell/table-cell.css (hunk 4 above)
Comment (MarcMatthiae) [reply packages/components/src/styles/table-cell/table-cell.css]: Moved to new sd-table component and updated to tw classes.

**Comment by MarcMatthiae** [comment]
> > > CleanShot 2023-09-13 at 12 40 44@2x
> > > Please make the background colors an enum, calling it `sd-table-cell--bg-...` to make clear, that this is only for the background.
> > > See e. g. CleanShot 2023-09-13 at 12 42 34@2x
> > 
> > 
> > I understood, that we provide this as part of the table-cell component and we also wrote that in the story: image
> > I don´t know if it is necessary, but at least we refined it like that.
> 
> Yeah, but we should document it properly. Just provide it as an own style. :)

Done :)

File: packages/components/src/styles/table-cell/table-cell.stories.ts [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+    const sortData: ('ascending' | 'descending' | 'none')[] = Array.from({ length: tableColumnCount }, () => 'none');
+    const headerData = Array.from({ length: tableColumnCount }, () => 'Header');
+    const tableData = Array.from({ length: tableRowCount }, () =>
+      Array.from({ length: tableColumnCount }, () => {
+        return args['default-slot'] ?? 'Lorem ipsum dolor sit amet.';
+      })
+    );
+
+    // Function to exchange the sort icons and set the aria-sort attribute in the sortable table after a click on the table header
+    const sortTable = (column: number) => {
```
[the remark this answers is not in the document]
Comment (MarcMatthiae) [reply packages/components/src/styles/table-cell/table-cell.stories.ts:66]: Done :)

File: packages/components/src/styles/table-cell/table-cell.stories.ts (hunk 1 above)
Comment (MarcMatthiae) [reply packages/components/src/styles/table-cell/table-cell.stories.ts]: #376

**Comment by coraliefeil** [comment]
Review done

**Comment by MarcMatthiae** [comment]
Main changes
- Removed "Sortable Table" sample
- Added "Simple Table" sample
- Added correct font color to sd-table-cell
- @coraliefeil Leider brauchen wir diese "default"-Farbe immer, haben wir am Freitag im Daily besprchen. Ich habe jetzt zumindest eingebaut, dass die Farbe in Storybook erstmal auf transparent spring, aber es gibt immer diesen (Default)-Eintrag.

**Review by karlbaumhauer — APPROVED** [review]

File: packages/components/src/styles/table-cell/table-cell.css [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  @apply border-solid border-neutral-400 border-y-[1px] border-x-0 text-black text-sm text-left p-4;
+
+  &--divider {
+    @apply border-r-[1px];
+  }
+
+  &--bg {
+    &-transparent {
+      @apply bg-transparent;
+    }
```
Comment (mariohamann) [inline packages/components/src/styles/table-cell/table-cell.css:11]: I think this should be part the default. Please remove it here and in the declaration and add bg-transparent to line 2.

File: packages/components/src/styles/table-cell/table-cell.declaration.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+  since: '1.12',
+  attributes: [
+    {
+      name: 'sd-table-cell--divider',
+      description: 'Displays a divider to the right.'
+    },
+    {
+      name: 'sd-table-cell--bg-...',
+      options: ['transparent', 'white', 'primary-100', 'neutral-100'],
+      description: 'Applies the selected backround-color to table cell.'
```
Comment (mariohamann) [inline packages/components/src/styles/table-cell/table-cell.declaration.ts:16]:
```suggestion
      description: 'Applies the selected background-color to table cell.'
```

**Review by karlbaumhauer — APPROVED** [review]

**Pull request merged**